src/vs/workbench/contrib/chat/browser/actions/chatContextActions.ts

```diff
@@ -270,17 +305,25 @@ class AttachContextAction extends Action2 {
 		const widgetService = accessor.get(IChatWidgetService);
 		const languageModelToolsService = accessor.get(ILanguageModelToolsService);
 		const quickChatService = accessor.get(IQuickChatService);
+		const clipboardService = accessor.get(IClipboardService);
+		const fileService = accessor.get(IFileService);
 		const context: { widget?: IChatWidget } | undefined = args[0];
 		const widget = context?.widget ?? widgetService.lastFocusedWidget;
 		if (!widget) {
 			return;
 		}
 
+		const imageData = await clipboardService.readImage();
+
 		const usedAgent = widget.parsedInput.parts.find(p => p instanceof ChatRequestAgentPart);
 		const slowSupported = usedAgent ? usedAgent.agent.metadata.supportsSlowVariables : true;
 		const quickPickItems: (IChatContextQuickPickItem | QuickPickItem)[] = [];
 		for (const variable of chatVariablesService.getVariables(widget.location)) {
 			if (variable.fullName && (!variable.isSlow || slowSupported)) {
+				if (variable.id === 'copilot.image' && !isImage(imageData)) {
```

Comment:
Should be able to remove this if you aren't registering a variable from the extension anymore

Reply:
I'm still creating an `imageResolver` on the extension side since i saw that most of the attachments were being attached to the quick pick in the extension. could this be done elsewhere, or would this be more in line with what we're already doing?

Reply:
jk this is a dumb question - directly adding it to quick pick here is fine as well i think